### ISSUE TITLE
Fuzzy autocomplete

### ISSUE THREAD

**Issue by paulcsmith** [msaraiva/atom-elixir#13]
First off, great work on the project! I'm not sure how autocomplete works in this project, but I wonder if it could use fuzzy searching (as an option) so `Phnx` would match `Phoenix`. I rely on this quite heavily from the default atom autocomplete. I wonder if this package could hook into atom's package to allow the fuzzy matching?

**Comment by msaraiva** [comment]
It would be a nice feature. I'll take a look at it after publishing the first release.

**Comment by paulcsmith** [comment]
Right on. Thanks for the awesome work!

**Comment by DorianGrey** [comment]
Just to mention an idea... 
If I didn't get that wrong, the completion suggestions are create by the alchemist server [here](https://github.com/msaraiva/atom-elixir/blob/master/lib/alchemist-server/lib/api/comp.exs#L63) resp. [here](https://github.com/msaraiva/atom-elixir/blob/master/lib/alchemist-server/lib/api/comp.exs#L69).
Wouldn't it be useful to use the built-in [jaro_distance](http://elixir-lang.org/docs/stable/elixir/String.html#jaro_distance/2) for fuzzy matching, as a replacement of the currently used startsWith?
If that is of any interest, I'd give it a shot a provide a pull request with it.

**Comment by msaraiva** [comment]
@DorianGrey I think it would take more than just replacing `startsWith` with `jaro_distance`. Users may want to keep the startsWith behaviour, so we would need to create a configuration for that and also change the API to pass that configuration to the server and than to all required functions, including the functions that lists modules and functions/macros. If you consider all the cases, including the configuration part, feel free to work on a PR.

**Comment by DorianGrey** [comment]
Hm... I'd suggest whether it might be useful to configure the autocompletion behavior to stay with "startsWith" (or maybe use a completely different strategy) depends on the results that the alternatives create. Maybe another fuzzy matching strategy based on Levenshtein-distance would be more suitable.

I'll take a closer look at these options next weekend.

**Comment by DorianGrey** [comment]
Hm... took some time to get Levenshtein to a good performance. The fuzzy completion works rather good on simple variables (although the decision rule should still be tuned a little bit). However, I didn't have in mind that starts_with is also used on a more higher level [here](https://github.com/msaraiva/atom-elixir/blob/master/lib/alchemist-server/lib/helpers/complete.exs#L368), which makes configuring the completion strategy more complicated then I expect. Will take some more time :(

**Comment by msaraiva** [comment]
@DorianGrey yeah, I think the whole autocomplete feature should eventually be refactored in order to be more flexible for extensions. Keep in mind that this code was originally designed to work with `iex`. I already made some customisation but it always fells hacky.

**Comment by sascha-wolf** [comment]
Any progess on this? I rely quite a lot on fuzzy autocomplete and find it quite irritating that it doesn't work with this plugin. Other than that, the plugin is awesome.
